tests/test_commodity_dictionary.py

```diff
@@ -22,3 +22,40 @@ def test_commod_names_outcomodity():
     assert ['fuel_outcommods'] == cd.get_commod_names(metadata["annotations"],
                                                       "outcommodity",
                                                       ":cycamore:Reactor")
+
+
+def test_build_facility_dictionary():
+    exp = {':agents:KFacility': (['in_commod'], ['out_commod']),
+           ':agents:NullInst': ([], []),
+           ':agents:NullRegion': ([], []),
+           ':agents:Predator': (['prey'], ['commod']),
+           ':agents:Prey': ([], ['commod']),
+           ':agents:Sink': (['in_commods'], []),
+           ':agents:Source': ([], ['commod']),
+           ':cycamore:DeployInst': ([], []),
+           ':cycamore:Enrichment': (['feed_commod'], ['product_commod',
+                                     'tails_commod']),
+           ':cycamore:FuelFab': (['fill_commods', 'fiss_commods',
+                                  'topup_commod'], ['outcommod']),
+           ':cycamore:GrowthRegion': ([], []),
+           ':cycamore:ManagerInst': ([], []),
+           ':cycamore:Mixer': ([], ['out_commod']),
+           ':cycamore:Reactor': (['fuel_incommods', 'pref_change_commods',
+                                  'recipe_change_commods'],
+                                 ['fuel_outcommods']),
+           ':cycamore:Separations': (['feed_commods'], ['leftover_commod',
+                                      'streams_']),
+           ':cycamore:Sink': (['in_commods'], []),
+           ':cycamore:Source': ([], ['outcommod']),
+           ':cycamore:Storage': (['in_commods'], ['out_commods'])}
+
+    assert cd.build_facility_dictionary(metadata, specs) == exp
+
+
+def test_build_facility_dictionary_subset():
+    archetypes = [':cycamore:Sink', ':cycamore:Source', ':cycamore:Storage']
+    exp = {':cycamore:Sink': (['in_commods'], []), 
+           ':cycamore:Source': ([], ['outcommod']),
+            ':cycamore:Storage': (['in_commods'], ['out_commods'])}
+
+    assert cd.build_facility_dictionary(metadata, archetypes) == exp
```

Comment:
best practice to have a new line at the end of the file

Reply:
```suggestion
    assert cd.build_facility_dictionary(metadata, archetypes) == exp
    
```